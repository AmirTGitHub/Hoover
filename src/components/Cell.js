import React from 'react';
import Hoover from '../assets/Hoover_01png'
import Dirt from '../assets/Dirt_01.png'

const Cell = (props) => {
    return (
        props.hoover !== props.position && props.dirt !== props.position ?
            <div className='cell' id={props.position} ></div>
            : props.hoover === props.position ? <img src={Hoover} className='cell' alt='a robotic hoover' />
                : props.dirt === props.position ? <img src={Dirt} className='cell' alt='an imaginary dirt' /> : null

    )
}

export default Cell