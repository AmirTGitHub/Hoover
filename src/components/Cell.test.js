import Cell from './Cell'

const setup = {
    hoover: 10,
    dirt: 37
}

const cellPropsForDiv = {
    position: 12,
}
it('render cell without hoover or dirt correctly', () => {
    const wrapper = render(
        <Cell {...cellPropsForDiv} />

    )
    expect(wrapper).toMatchSnapshot()
})

const cellPropsForHoover = { ...setup, position: setup.hoover }
it('render the cell containing hoover position', () => {
    const wrapper = render(
        <Cell {...cellPropsForHoover} />

    )
    expect(wrapper).toMatchSnapshot()
})
const cellPropsForDirt = { ...setup, position: setup.dirt };

it('render the cell containing dirt correctly', () => {
    const wrapper = render(
        <Cell {...cellPropsForDirt} />

    )
    expect(wrapper).toMatchSnapshot()
})