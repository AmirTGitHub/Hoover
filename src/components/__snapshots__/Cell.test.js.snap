// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render cell without hoover or dirt correctly 1`] = `
<div
  class="cell"
  id="12"
>
  
</div>
`;

exports[`render the cell containing dirt correctly 1`] = `
<img
  alt="an imaginary dirt"
  class="cell"
  src="Dirt_01.png"
>
  
</img>
`;

exports[`render the cell containing hoover position 1`] = `
<img
  alt="a robotic hoover"
  class="cell"
  src="Hoover_01png"
>
  
</img>
`;
